CMakeLists.txt

```diff
@@ -549,39 +549,36 @@ else()
     endif()
   endif()
 
-  option(NO_AES "Explicitly disable AES support" ${NO_AES})
+# Detect and set crypto extensions for specific platforms
 
-  if(NO_AES)
-    message(STATUS "AES support explicitly disabled")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNO_AES")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DNO_AES")
-  elseif(NOT ARM AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X)
-    message(STATUS "AES support enabled")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maes")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -maes")
-  elseif(PPC64LE OR PPC64 OR PPC)
+  if(PPC64LE OR PPC64 OR PPC)
     message(STATUS "AES support not available on POWER")
+    option(NO_AES "Explicitly disable AES support" ${NO_AES})
   elseif(S390X)
     message(STATUS "AES support not available on s390x")
+    option(NO_AES "Explicitly disable AES support" ${NO_AES})
   elseif(ARM6)
     message(STATUS "AES support not available on ARMv6")
+    option(NO_AES "Explicitly disable AES support" ${NO_AES})
   elseif(ARM7)
     message(STATUS "AES support not available on ARMv7")
+    option(NO_AES "Explicitly disable AES support" ${NO_AES})
   elseif(ARM8)
-    CHECK_CXX_ACCEPTS_FLAG("-march=${ARCH}+crypto" ARCH_PLUS_CRYPTO)
+    CHECK_CXX_ACCEPTS_FLAG("-march=${ARCH}+aes" ARCH_PLUS_CRYPTO)
     if(ARCH_PLUS_CRYPTO)
       message(STATUS "Crypto extensions enabled for ARMv8")
-      set(ARCH_FLAG "-march=${ARCH}+crypto")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=${ARCH}+aes")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${ARCH}+aes")
     else()
       message(STATUS "Crypto extensions unavailable on your ARMv8 device")
+      option(NO_AES "Explicitly disable AES support" ${NO_AES})
     endif()
-  else()
-    message(STATUS "AES support disabled")
+  elseif(NOT ARM AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X)
```

Comment:
No reason to have `AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X` here